modules/core/client/app/init.js

```diff
@@ -1,80 +1,45 @@
-'use strict';
+(function (app) {
+  'use strict';
 
-// Start by defining the main module and adding the module dependencies
-angular.module(ApplicationConfiguration.applicationModuleName, ApplicationConfiguration.applicationModuleVendorDependencies);
+  // Start by defining the main module and adding the module dependencies
+  angular
+    .module(app.applicationModuleName, app.applicationModuleVendorDependencies);
 
-// Setting HTML5 Location Mode
-angular.module(ApplicationConfiguration.applicationModuleName).config(['$locationProvider', '$httpProvider',
-  function ($locationProvider, $httpProvider) {
+  // Setting HTML5 Location Mode
+  angular
+    .module(app.applicationModuleName)
+    .config(bootstrapConfig);
+
+  function bootstrapConfig($locationProvider, $httpProvider) {
     $locationProvider.html5Mode(true).hashPrefix('!');
 
     $httpProvider.interceptors.push('authInterceptor');
   }
-]);
-
-angular.module(ApplicationConfiguration.applicationModuleName).run(function ($rootScope, $state, Authentication) {
-
-  // Check authentication before changing state
-  $rootScope.$on('$stateChangeStart', function (event, toState, toParams, fromState, fromParams) {
-    if (toState.data && toState.data.roles && toState.data.roles.length > 0) {
-      var allowed = false;
-      toState.data.roles.forEach(function (role) {
-        if ((role === 'guest') || (Authentication.user && Authentication.user.roles !== undefined && Authentication.user.roles.indexOf(role) !== -1)) {
-          allowed = true;
-          return true;
-        }
-      });
 
-      if (!allowed) {
-        event.preventDefault();
-        if (Authentication.user !== undefined && typeof Authentication.user === 'object') {
-          $state.go('forbidden');
-        } else {
-          $state.go('authentication.signin').then(function () {
-            storePreviousState(toState, toParams);
-          });
-        }
+  bootstrapConfig.$inject = ['$locationProvider', '$httpProvider'];
+
+  // Then define the init function for starting up the application
+  angular.element(document).ready(init);
+
+  function init() {
+    // Fixing facebook bug with redirect
+    if (window.location.hash && window.location.hash === '#_=_') {
+      if (window.history && history.pushState) {
+        window.history.pushState('', document.title, window.location.pathname);
+      } else {
+        // Prevent scrolling by storing the page's current scroll offset
+        var scroll = {
+          top: document.body.scrollTop,
+          left: document.body.scrollLeft
+        };
+        window.location.hash = '';
```

Comment:
Have you tried it yourself with any success or does this just make sense in your head?